### PR TITLE
Add remote.Digest() to upload a manifest via digest

### PR DESCRIPTION
**Pull request by deitch**
As discussed with @jonjohnsonjr . We already have [remote.Tag](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/v1/remote#Tag) which uploads a manifest and references it with a tag. However, there also is the ability in the API to upload a manifest and just reference it via its digest, just like any other blob. Eventually, the manifest should be referenced by another manifest or a tag, but those are independent.

### CODE REVIEW

File: pkg/v1/remote/write.go [hunk 1]
```diff
@@ -849,27 +849,48 @@ func WriteLayer(repo name.Repository, layer v1.Layer, options ...Option) (rerr e
...
+// Notable implementations of Taggable are v1.Image, v1.ImageIndex, and
+// remote.Descriptor.
+//
+// If t implements MediaType, we will use that for the Content-Type, otherwise
+// we will default to types.DockerManifestSchema2.
+//
+// Digest does not attempt to write anything other than the manifest, so callers
+// should ensure that all blobs or manifests that are referenced by t exist
+// in the target registry.
+func Digest(digest name.Digest, t Taggable, options ...Option) error {
```
Comment (jonjohnsonjr) [inline pkg/v1/remote/write.go:866]: I'd lean towards just `remote.Put` for this, I think. The existing `remote.Tag` is using "tag" as a verb here, not as a noun, so calling this `Digest` doesn't make a ton of sense.

`remote.Put` would also mirror `remote.Get`, `remote.Head`, and `remote.Delete`.

It should take a `name.Reference` as well, IMO (basically just exposing `manifestReference` as `Put`).

**Review by jonjohnsonjr — CHANGES_REQUESTED** [review]

File: pkg/v1/remote/write.go (hunk 1 above)
Comment (deitch) [reply pkg/v1/remote/write.go]: Done. Take a look.

**Review by jonjohnsonjr — APPROVED** [review]

**Comment by jonjohnsonjr** [comment]
LGTM, a test would be cool, but just rebasing also works :)

**Comment by deitch** [comment]
Rebased.

**Comment by jonjohnsonjr** [comment]
Thanks!

**Pull request merged**